x-pack/legacy/plugins/apm/server/routes/create_api/index.ts

```diff
@@ -51,74 +64,111 @@ export function createApi() {
           body: bodyRt && 'props' in bodyRt ? t.exact(bodyRt) : fallbackBodyRt
         };
 
-        server.route(
-          merge(
-            {
-              options: {
-                tags: ['access:apm']
-              },
-              method: 'GET'
+        router[routerMethod](
+          {
+            path,
+            options: {
+              tags: ['access:apm']
             },
-            route,
-            {
-              handler: (request: Request, h: ResponseToolkit) => {
-                const paramMap = {
-                  path: request.params,
-                  body: request.payload,
-                  query: request.query
-                };
-
-                const parsedParams = (Object.keys(rts) as Array<
-                  keyof typeof rts
-                >).reduce(
-                  (acc, key) => {
-                    const codec = rts[key];
-                    const value = paramMap[key];
-
-                    const result = codec.decode(value);
-
-                    if (isLeft(result)) {
-                      throw Boom.badRequest(PathReporter.report(result)[0]);
-                    }
-
-                    const strippedKeys = difference(
-                      Object.keys(value || {}),
-                      Object.keys(result.right || {})
+            validate: {
+              ...(routerMethod === 'get'
+                ? {}
+                : { body: schema.object({}, { allowUnknowns: true }) }),
```

Comment:
As long as you ensure that you're always validating route inputs it's fine to short-circuit NP's route validation here. We have an open issue to discuss not making validation tied so tightly to kbn/config-schema and we hope to make a decision about it in the next week.

Reply:
just read about the decision, great news for us 😬